File: src/main/apache/src/app/model/paginated-response.ts
export class PaginatedResponse<T> {

  private _data: T[];

  private _totalCount: number;

  get data(): T[] {
    return this._data;
  }

  set data(value: T[]) {
    this._data = value;
  }

  get totalCount(): number {
    return this._totalCount;
  }

  set totalCount(value: number) {
    this._totalCount = value;
  }
}

File: src/main/java/cloud/martinodutto/tpt/pagination/PaginatedResponse.java
package cloud.martinodutto.tpt.pagination;

import java.util.List;

public class PaginatedResponse<T> {

    private List<T> data;

    private long totalCount;

    public PaginatedResponse(List<T> data, long totalCount) {
        this.data = data;
        this.totalCount = totalCount;
    }

    public List<T> getData() {
        return data;
    }

    public void setData(List<T> data) {
        this.data = data;
    }

    public long getTotalCount() {
        return totalCount;
    }

    public void setTotalCount(long totalCount) {
        this.totalCount = totalCount;
    }

    @Override
    public String toString() {
        return "PaginatedResponse{" +
                "data=" + data +
                ", totalCount=" + totalCount +
                '}';
    }
}
